crates/astria-account-monitor/src/main.rs

```diff
@@ -0,0 +1,67 @@
+use std::process::ExitCode;
+
+use astria_account_monitor::{
+    config::Config,
+    AccountMonitor,
+    BUILD_INFO,
+};
+use astria_eyre::eyre::WrapErr as _;
+use tokio::signal::unix::{
+    signal,
+    SignalKind,
+};
+
+#[tokio::main]
+async fn main() -> ExitCode {
+    astria_eyre::install().expect("astria eyre hook must be the first hook installed");
+
+    eprintln!(
+        "{}",
+        serde_json::to_string(&BUILD_INFO)
+            .expect("build info is serializable because it contains only unicode fields")
+    );
+
+    let cfg: Config = config::get().expect("failed to read configuration");
+
+    let (metrics, _telemetry_guard) = match telemetry::configure()
+        .set_no_otel(cfg.no_otel)
+        .set_force_stdout(cfg.force_stdout)
+        .set_filter_directives(&cfg.log)
+        .set_metrics(&cfg.metrics_http_listener_addr, env!("CARGO_PKG_NAME"))
+        .try_init(&cfg)
+        .wrap_err("failed to setup telemetry")
+    {
+        Err(error) => {
+            eprintln!("failed to setup telemetry:\n {error}");
+            return ExitCode::FAILURE;
+        }
+        Ok(telemetry_conf) => telemetry_conf,
+    };
+
+    let mut account_monitor = match AccountMonitor::spawn(cfg, metrics) {
+        Err(error) => {
+            eprintln!("failed to start account monitor:\n {error}");
+            return ExitCode::FAILURE;
+        }
+        Ok(account_monitor) => account_monitor,
+    };
+
+    let mut sigterm = signal(SignalKind::terminate())
+        .expect("setting a SIGTERM listener should always work on Unix");
+
+    tokio::select! {
+        _ = sigterm.recv() => {
+            account_monitor.shutdown();
```

Comment:
Right now, if we receive a shutdown signal this cancels the service, but then immediately exits. This means that if we ever have some cleanup logic that needs to run, it will not run.

Suggestion for an easy fix: turn `AccountMonitor::shutdown` into a `async fn` and then let it await itself, i.e. something like:

```rust
impl AccountMonitor {
    pub async fn shutdown(self) -> eyre::Result<()> {
        self.shutdown_token.cancel();
        self.await // using the fact that it's a future
    }
}
```

You can then have both `select!` arms returns the same information (so you don't duplicate code).